File: src/002/lua/002.lua
-- non-recursive solution
function fib(n)
  f = {}
  f[0], f[1] = 1, 1
  for i = 2, n do
    f[i] = f[i - 1] + f[i - 2]
  end
  return f[n]
end

local i, sum, f = 0, 0, 0

while f < 4000000 do
  f = fib(i)
  if f % 2 == 0 then
    sum = sum + f
  end
  i = i + 1
end

print(sum)

File: src/002/python/002.py
import functools

# recursive solution but
# using memoization with lru_cache()

@functools.lru_cache()
def fib(n):
    if n in (0, 1):
        return 1
    else:
        return fib(n-1) + fib(n-2)

c,f,s=0,0,0
while f < 4000000:
    f=fib(c)
    c += 1
    if (f % 2) == 0:
        s += f

print(s)
